### DISCUSSION TITLE
Exiting from a command that was specified in the layout kdl file

### DISCUSSION THREAD

**Discussion by Eloquencere** [zellij-org/zellij#3870] [Q&A] [answered]
I had lazygit running as a command specified in a layout.kdl file, but when I exit from the session, I am faced with whether I want to rerun the command, drop the shell, or exit. I prefer not to be asked this question and want to return to the shell, is it possible?

**Comment by Lockszmith-GH** [comment]
That is [Session Resurrection](https://zellij.dev/documentation/session-resurrection.html), which is on by default.

Also see: #2919

**Comment by Eloquencere** [comment]
Sorry, I think I didn't explain it correctly, I meant the lazygit session, not zellij. When I quit a command that was specified as a `command` in the layout file, I am faced with the 3 options I described above, but say that I don't want to be asked that for that pane how would I be able to do it?

**Reply by Lockszmith-GH** [reply]
Maybe you're looking for [close_on_exit](https://zellij.dev/documentation/creating-a-layout#close_on_exit)?

If not, please describe the workflow you are looking for vs the workflow that is happening today, and I'm sure we can figure it out.

(BTW, note that each comment, opens a thread, and you can "Write a reply..." in the thread vs "Add your answer here...")

**Reply by Eloquencere** [reply]
I think `close_on_exit` is pretty close to what I was looking for but when I implemented it on my layout file, there's not change in how zellij behaves.

**Reply by Lockszmith-GH** [reply]
Can you please describe the workflow you are looking for vs the workflow that is happening today?

**Reply by Eloquencere** [reply]
Apologies, my bad. So, say in the `kdl` file I declare that I want `yazi` to run on one of my panes (which works just fine) but say that I quit out of `yazi` and I want to get back to my shell, I do that with the `q` key now, I am faced with three options - to re-run the command, to drop to shell and to quit from the shell. Say that my answer to this question will always be to drop to shell, how can I tell `zellij` that?

**Reply by Lockszmith-GH** [reply, marked as answer]
I see... one way I can think of, is to not run `yazi`, but instead run it in a script that 'exits to the shell', from the command line it will look something like:

```shell
sh -c 'yazi; exec /bin/bash -l'
```

so this will run `sh` with the command line `yazi; exec /bin/bash -l`, which will first run `yazi`, then when it drops from `yazi`, it will replace the existing process with `/bin/bash -l` which will open an interactive shell.

**Reply by Eloquencere** [reply]
Thanks for that reply, but I feel it to be very primitive as compared to something like `tmuxp` maybe it's something can be worked on in the future. Also it's a really sad thing that only one command can be issued in the layout file, but for situations like when I would like a command to be run on a different folder than the default one I *need* to execute a `cd` and then my command, making the current options provided by `zellij` quite incomplete.

**Reply by Lockszmith-GH** [reply]
Specifically to your gripe, the Current Working Dir for a command _can_ be set, [see cwd](https://zellij.dev/documentation/keybindings-possible-actions.html?highlight=cwd#run).
BTW: If you search the discussions and PRs in this repo for `cwd` you will see that painstakingly attention is being placed on it, it's not a trivial afterthought.

**Reply by Eloquencere** [reply]
I have another question, I want a layout to look something like this

-----------------
|  Pane1 | Pane2  |
| ----------------  |
|       Pane3         |
-----------------

And my layout file looks something like this:
```
    tab name="Dev" focus=true {
        pane name="Pane1" focus=true {
            split_direction "vertical"
            size "70%"
        }
        pane name="Pane3" {
            split_direction "horizontal"
            cwd "$PROJECT_DIR"
            size "30%"
        }
        pane name="Pane2" {
            split_direction "vertical"
            size "70%"
        }
    }
```
This isn't giving me that. Could you guide me on how to do it?

**Reply by Lockszmith-GH** [reply]
I've done that before, let me check and get back to you

**Reply by Lockszmith-GH** [reply]
yes, here it is:
Tabs and Pane are both containers, their default split-direction is "horizontal", so what you described are 3 horizontal panes.

What you want is:
1. Horizontal pane containing 2 vertically split panes
2. Another Horizontal pane

I created the following file `~/.config/zellij/layouts/dev.kdl`:

```kdl
layout {
    cwd "$HOME"
    tab name="Dev" focus=true hide_floating_panes=true {
        pane size=1 borderless=true {
            plugin location="zellij:tab-bar"
        }
        pane size="30%" split_direction="vertical" {
            pane size="70%" focus=true
            pane size="30%"
        }
        pane focus=true size="70%" {
        cwd "$PROJECT_DIR"
        }
        pane size=1 borderless=true {
            plugin location="zellij:status-bar"
        }
    }
}
```

Launching the layout above, gave me the layout that matches what you described you wanted.
```console
$ zellij --layout=$HOME/.config/zellij/layouts/dev.kdl
```

**Reply by Eloquencere** [reply]
Yup, that worked beautifully, your template is perfect!

**Comment by Lockszmith-GH** [comment]
Opening a separate thread, so that I can delete it if becomes a 'thing of it's own' - which I don't intend it to be.

In response to [your reply](https://github.com/zellij-org/zellij/discussions/3870#discussioncomment-11646430), **emphasis** added by me:
> Thanks for that reply, but **I feel** it to be **very primitive** as compared to something like `tmuxp` maybe it's something can be worked on in the future. Also it's **a really sad thing** that only one command can be issued in the layout file, but for situations like when **I would like** a command to be run on a different folder than the default one **I _need_ to** execute a `cd` and then my command, making the current options provided by `zellij` quite incomplete.

I may be the asshole here for being triggered by it, and I wrote a whole piece about "how we should behave", then decided against it, as to not hijack this conversation too much - as it would just make things worse for everyone.

It feels like you're trying to insult the tool and/or it's development by the use of words like "primitive", "really sad".
I'm sorry Zellij is overburdening you.

While I doubt you felt your last comment was toxic in any way, I'd like you to take note how it _can_ be perceived.

I would recommend you read [this thread](https://github.com/twpayne/chezmoi/discussions/2819) from another community and the articles linked there.

**Reply by Eloquencere** [reply]
I'm sorry if my words seemed a bit harsh; it wasn't my intention for them to be so. I love Zellij, and I use it all the time; I even switched over from Tmux because of the potential I saw in it, and it gives me great pride in contributing constructively to its development. I appreciate your efforts towards good etiquette; it is entirely my fault for the poor choice of words. I honestly don't know why I said that, considering how helpful you have been to me all along; I hope I am worthy of your forgiveness.

**Reply by Lockszmith-GH** [reply]
🤣 We are all good, and sorry I took it too far. I have been perceived as a douchebag numerous times, probably because I care too much, so I've learned to pay close attention to language.

And no forgiveness of mine is needed, you are not at fault (besides, I'm of little importance).
I'm glad it was mostly a misunderstanding on **my** part.

It's all about responsibility anyway, I take full responsibility here for overreacting.

**Reply by Eloquencere** [reply]
Thank you so much, I am really grateful!
